### ISSUE TITLE
Feature request for an integrated diff feature.

### ISSUE THREAD

**Issue by abvgdeyoj** [exiftool/exiftool#278]
[diff simply doesn't cut it.](https://exiftool.org/forum/index.php?topic=3276)

My own usecase is to see what information gets dropped by cjxl by comparing the original JPEG and the result, and store it in a separate XML (-X) file if there is any.

**Comment by boardhead** [comment]
I've been working on this, and currently have a test version that gives output like this:

```
% exiftool a.jpg -diff b.jpg
diff < a.jpg > b.jpg
   ExifTool       (2 same tags)
   System         (2 same tags)
< [System]        FileName: a.jpg
< [System]        FileModifyDate: 2024:10:16 15:33:51-04:00
< [System]        FileAccessDate: 2024:10:16 15:33:53-04:00
< [System]        FileInodeChangeDate: 2024:10:16 15:33:51-04:00
< [System]        FilePermissions: -rw-------
> [System]        FileName: b.jpg
> [System]        FileModifyDate: 2024:10:16 15:33:55-04:00
> [System]        FileAccessDate: 2024:10:16 15:33:56-04:00
> [System]        FileInodeChangeDate: 2024:10:16 15:33:55-04:00
> [System]        FilePermissions: -rw-r--r--
   File           (11 same tags)
   IFD0           (5 same tags)
> [IFD0]          Artist: me
   IPTC           (2 same tags)
   Photoshop      (1 same tag)
   XMP-x          (1 same tag)
   XMP-dc         (1 same tag)
< [XMP-dc]        Subject: test a
> [XMP-dc]        Subject: test b
   JUMBF          (42 same tags)
   Jpeg2000       (9 same tags)
   CBOR           (56 same tags)
   JSON           (12 same tags)
   JFIF           (4 same tags)
   Composite      (5 same tags)
```

- Phil

**Comment by boardhead** [comment]
I've reworked it a bit and am happy with the result:

```
% exiftool a.jpg -diff b.jpg --system:all
======== diff < a.jpg > b.jpg
> [File]          Exif Byte Order                 : Big-endian (Motorola, MM)
< [XMP-dc]        Subject                         : test a
>                                                   test b
> [Composite]     Physical Image Size             : 0.1x0.1 inches
> [IFD0]          Y Resolution                    : 72
> [IFD0]          Resolution Unit                 : inches
> [IFD0]          Artist                          : me
> [IFD0]          Y Cb Cr Positioning             : Centered
```

In the above command I am ignoring the System tags with `--system:all`.  And now the "same tag" lines are output only with the `-v` option.

I'm looking to include this feature in version 12.99

- Phil

**Comment by boardhead** [comment]
I've just released version 12.99 with this new feature.

- Phil
